pandas/tests/indexing/test_indexing.py

```diff
@@ -542,6 +570,23 @@ def test_astype_assignment_with_dups(self):
         # result = df.get_dtype_counts().sort_index()
         # expected = Series({'float64': 2, 'object': 1}).sort_index()
 
+    @pytest.mark.parametrize("index,val,expected", [
+        # Checking if np.inf in Int64Index should not cause an OverflowError
+        (pd.Int64Index([0, 1, 2]), np.inf, False),
+        (pd.Int64Index([0, 1, 2]), np.nan, False),
+        (pd.UInt64Index([0, 1, 2]), np.inf, False),
+        (pd.UInt64Index([0, 1, 2]), np.nan, False),
+        # Check contains correctly checks for np.inf
+        (pd.Index([0, 1, 2, np.inf]), np.inf, True),
+        (pd.Index([0, 1, 2, np.nan]), np.nan, True),
+        (pd.Index([0, 1, 2, np.inf]), np.nan, False),
+        (pd.Index([0, 1, 2, np.nan]), np.inf, False),
+    ])
```

Comment:
Nice!  Lots of good test cases.

However, I'd like to split these in two tests based on the expected result.  This is because statements like "assert result is True" are less idiomatic.  Rather, we should be saying "assert val in index" OR "assert val not in index."

It also means one fewer parameter in your `pytest` parametrization. :smile: